pkg/controller/utils/utils.go

```diff
@@ -647,7 +647,8 @@ func createPredicateForObject(objMeta metav1.Object) predicate.Predicate {
 			return e.Object.GetNamespace() == objMeta.GetNamespace() || objMeta.GetNamespace() == ""
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			generationChanged := e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration()
+			// React if the generation isn't set (== 0).
```

Comment:
Could you add a comment saying that not all objects use/have a generation and that therefore, this addition is required?